File: src/simulation/culture/group/intergroup/Relation.java
package simulation.culture.group.intergroup;

import simulation.culture.group.Group;

public class Relation {
    public Group owner;
    public Group other;

    double positive = 0;

    private Relation pair;

    public Relation(Group owner, Group other) {
        this.owner = owner;
        this.other = other;
    }

    public double getPositive() {
        return positive;
    }

    public void setPair(Relation pair) {
        this.pair = pair;
    }

    public void setPositive(double positive) {
        this.positive = positive;
        if (this.positive > 1) positive = 1;
        if (this.positive < -1) positive = -1;
    }

    public void increaseRelationsTwoSided(double delta) {
        setPositive(positive + delta);
        pair.setPositive(pair.positive + delta);
    }

    @Override
    public String toString() {
        return String.format("%s is %f", other.name, positive);
    }
}

File: src/simulation/culture/group/intergroup/Relation.kt
package simulation.culture.group.intergroup

import simulation.culture.group.centers.Group

class Relation(var owner: Group, var other: Group, var status: Status = Status.None) {
    var positive = 0.0
        set(value) {
            field = value
            if (positive > 1) field = 1.0
            if (positive < -1) field = -1.0
            if (field <= 0.00001) status = Status.War
        }

    var positiveInteractions = 0.0

    var pair: Relation = this

    override fun toString() = "${other.name} is $positive"

    val normalized: Double
        get() = (positive + 1) / 2
}

enum class Status {
    None,
    War
}
